app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -0,0 +1,37 @@
+module OpenidConnect
+  class AuthorizationController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def index
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+
+      success = @authorize_form.valid?
+      analytics.track_event(Analytics::OPENID_CONNECT_REQUEST_AUTHORIZATION,
+                            success: success,
+                            client_id: @authorize_form.client_id,
+                            errors: @authorize_form.errors.full_messages)
+
+      render(success ? :index : :error)
+    end
+
+    def create
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+
+      result = @authorize_form.submit(current_user)
+      analytics.track_event(Analytics::OPENID_CONNECT_ALLOW, result.except(:redirect_uri))
+
```

Comment:
The `redirect_uri` contains the `code` param which is sensitive as it could be used to create a session as somebody else? It's not trivial (because the client requests need to be signed) but I figured it would be better to err on the side of privacy? Based on the `client_id` we know where we *should* be redirecting to